metals/src/main/scala/scala/meta/internal/metals/Indexer.scala

```diff
@@ -430,9 +431,17 @@ final case class Indexer(
         val reluri = source.toIdeallyRelativeURI(sourceItem)
         val input = sourceToIndex0.toInput
         val symbols = ArrayBuffer.empty[WorkspaceSymbolInformation]
+        val methodSymbols = ArrayBuffer.empty[WorkspaceSymbolInformation]
         SemanticdbDefinition.foreach(input, dialect) {
           case SemanticdbDefinition(info, occ, owner) =>
-            if (WorkspaceSymbolProvider.isRelevantKind(info.kind)) {
+            // TODO: Do not index (extension) METHOD, they will be indexed later
+            // we index methods for auto-import missing extension methods feature for now
+            // but those feature should use methodSymbols
+            if (
+              WorkspaceSymbolProvider.isRelevantKind(
+                info.kind
+              ) || info.kind == Kind.METHOD
```

Comment:
```suggestion
              )
```
do we want to search for methods? I don't think we normally index it.

Reply:
Nice catch! However, we have to index methods for now because

- now [ScalaToplevelMtags emit extension methods with Kind.Method](https://github.com/scalameta/metals/blob/57b05f3f8a32d7ce109a2f1011ae54f1660b3956/mtags/src/main/scala/scala/meta/internal/mtags/ScalaToplevelMtags.scala#L158-L164)
- We have to store them in `inWorkspace` because auto-import missing extension feature still uses `inWorkspace` index.
  - we can migrate auto-import missing extension feature to use `inWorkspaceMethod` (that is defined in this PR) later, then we can remove this
  - I'll open an issue :)

Reply:
https://github.com/scalameta/scalameta/issues/2799

Reply:
oops this one https://github.com/scalameta/metals/issues/4212